_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -841,27 +841,36 @@ public function update_data( $request ) {
 					break;
 
 				case 'wordpress_api_key':
+					if ( ! class_exists( 'Akismet' ) ) {
+						if ( ! file_exists( WP_PLUGIN_DIR . '/akismet/class.akismet.php' ) ) {
+							// Akismet is not installed as an mu plugin or in the plugins directory.
+							$error   = esc_html__( 'Please install Akismet.', 'jetpack' );
+							$updated = false;
+							break;
+						}
 
-					if ( ! file_exists( WP_PLUGIN_DIR . '/akismet/class.akismet.php' ) ) {
-						$error = esc_html__( 'Please install Akismet.', 'jetpack' );
-						$updated = false;
-						break;
-					}
-
-					if ( ! defined( 'AKISMET_VERSION' ) ) {
-						$error = esc_html__( 'Please activate Akismet.', 'jetpack' );
+						// Akismet is installed in the plugins directory but not active.
+						$error   = esc_html__( 'Please activate Akismet.', 'jetpack' );
 						$updated = false;
 						break;
 					}
 
-					// Allow to clear the API key field
+					// Allow to clear the API key field.
 					if ( '' === $value ) {
 						$updated = get_option( $option ) != $value ? update_option( $option, $value ) : true;
 						break;
 					}
 
-					require_once WP_PLUGIN_DIR . '/akismet/class.akismet.php';
-					require_once WP_PLUGIN_DIR . '/akismet/class.akismet-admin.php';
+					if ( ! defined( 'AKISMET__PLUGIN_DIR' ) ) {
+						return new WP_Error(
```

Comment:
Instead of returning a `WP_Error` here, would we be able to set `$error`, and `$updated` to false, like we do above?